Compliance Operator v0.1.57 release notes and enhancements added

sheriff-rh opened the pull request:
Version(s):
4.8+

Issue:
https://issues.redhat.com/browse/CMP-1552

Link to docs preview:
[OpenShift Compliance Operator 0.1.57 release notes](http://file.rdu.redhat.com/antaylor/co-0.1.55-rns/security/compliance_operator/compliance-operator-release-notes.html#compliance-operator-release-notes-0-1-55)
[Increasing Compliance Operator resource limits](http://file.rdu.redhat.com/antaylor/co-0.1.55-rns/security/compliance_operator/compliance-operator-troubleshooting.html#compliance-increasing-operator-limits)
[Setting PriorityClass for ScanSetting scans
](http://file.rdu.redhat.com/antaylor/co-0.1.55-rns/security/compliance_operator/compliance-operator-advanced.html#compliance-priorityclass_compliance-advanced)
[ScanSetting Custom Resource](http://file.rdu.redhat.com/antaylor/co-0.1.55-rns/security/compliance_operator/compliance-scans.html#compliance-scansetting-cr)
[Applying resource requests and limits
](http://file.rdu.redhat.com/antaylor/co-0.1.55-rns/security/compliance_operator/compliance-scans.html#compliance-applying-resource-requests-and-limits)
[Scheduling Pods with resource requests](http://file.rdu.redhat.com/antaylor/co-0.1.55-rns/security/compliance_operator/compliance-scans.html#compliance-scheduling-pods-with-resource-requests)
[Evaluating KubeletConfig rules against default configuration values
](http://file.rdu.redhat.com/antaylor/co-0.1.55-rns/security/compliance_operator/compliance-operator-remediation.html#compliance-evaluate-kubeletconfig-rules)
[Creating Custom node pools
](http://file.rdu.redhat.com/antaylor/co-0.1.55-rns/security/compliance_operator/compliance-operator-remediation.html#compliance-custom-node-pools)
[Remediating KubeletConfig sub pools
](http://file.rdu.redhat.com/antaylor/co-0.1.55-rns/security/compliance_operator/compliance-operator-remediation.html#compliance-kubeletconfig-sub-pool-remediation)

Wiharris commented on modules/compliance-increasing-operator-limits.adoc:
```diff
@@ -0,0 +1,75 @@
+
+[NOTE]
+====
+Resource Constraints applied in this process will overwrite existing resource constraints.
```
```suggestion
Resource Constraints applied in this process will overwrite the existing resource constraints.
```

Wiharris commented on modules/compliance-applying-resource-requests-and-limits.adoc:
```diff
@@ -0,0 +1,18 @@
+
+[IMPORTANT]
+====
+A container may or may not be allowed to exceed its CPU limit for extended periods. Container runtimes does not stop Pods or containers for excessive CPU usage. To determine whether a container cannot be scheduled or is being killed due to resource limits, see xref:../../security/compliance_operator/compliance-operator-troubleshooting.adoc#compliance-operator-troubleshooting[Troubleshooting the C ...
```
```suggestion
A container may not exceed its CPU limit for extended periods. Container run times do not stop Pods or containers for excessive CPU usage. To determine whether a container cannot be scheduled or is being killed due to resource limits, see xref:../../security/compliance_operator/compliance-operator-troubleshooting.adoc#compliance-operator-troubleshooting[Troubleshooting the Compliance Operator].
```

Wiharris commented on modules/compliance-custom-node-pools.adoc:
```diff
@@ -0,0 +1,89 @@
+
+The Compliance Operator does not maintain a copy of each node pool configuration. The Compliance Operator aggregates consistent configuration options for all nodes within a single node pool into one copy of the configuration file. The Compliance Operator then uses the configuration file for a particular node pool to evaluate rules against nodes within that pool.
+
+If your cluster uses custom node pools outside the default `worker` and `master` node pools, you will need to supply additional variables to ensure the Compliance Operator aggregates a configuration file for that node pool.
```
Not sure if this could be a note or could be left as is.

sheriff-rh replied:
I think I'm going to leave it as-is since we are describing the process. Might revisit that later prior to merge, though. Good looking out!

Wiharris commented on modules/compliance-custom-node-pools.adoc:
```diff
@@ -0,0 +1,89 @@
+  name: default
+----
+
+The Compliance Operator checks the runtime `KubeletConfig` through the Kubernetes `Node/Proxy` object, and then uses variables `ocp-var-role-master` and `ocp-var-role-worker` to determine the nodes it performs the check against. In the `ComplianceCheckResult`, the `KubeletConfig` rules will be shown as `ocp4-cis-kubelet-*`. The scan only passes if all selected nodes pass the check.
```
```suggestion
The Compliance Operator checks the runtime `KubeletConfig` through the Kubernetes `Node/Proxy` object, and then uses variables such as `ocp-var-role-master` and `ocp-var-role-worker` to determine the nodes it performs the check against. In the `ComplianceCheckResult`, the `KubeletConfig` rules will be shown as `ocp4-cis-kubelet-*`. The scan only passes if all selected nodes pass this check.
```

Wiharris commented on modules/compliance-evaluate-kubeletconfig-rules.adoc:
```diff
@@ -0,0 +1,9 @@
+[id="compliance-evaluate-kubeletconfig-rules"]
+= Evaluating `KubeletConfig` rules against default configuration values
+
+Kubernetes infrastructure may contain incomplete configuration files at run time, and nodes will assume default configuration values for missing configuration options. Some configuration options can be passed as command line arguments. As a result, the Compliance Operator cannot verify the configuration file on the node is complete or correct, as it may be missing options used in the rule checks.
```
```suggestion
Kubernetes infrastructure may contain incomplete configuration files at run time, and nodes will assume default configuration values for missing configuration options. Some configuration options can be passed as command line arguments. As a result, the Compliance Operator cannot verify if the configuration file on the node is complete, as it may be missing options used in the rule checks.
```

Wiharris commented on modules/compliance-increasing-operator-limits.adoc:
```diff
@@ -0,0 +1,75 @@
+[id="compliance-increasing-operator-limits"]
+= Increasing Compliance Operator resource limits
+
+In some cases, the Compliance Operator might require more memory than the default limits allow. If the Operator has been installed through OLM, the best way to mitigate issues is to set custom resource limits.
```
```suggestion
If the Operator has been installed through OLM the, Compliance Operator may require more memory than the default limits allow. The best way to mitigate this particular issue is to set custom resource limits.
```

Wiharris reviewed (commented):
LGTM! Left some minor comments! :)

Wiharris commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -13,6 +13,52 @@ These release notes track the development of the Compliance Operator in the {pro
+
+* Previously, the Compliance Operator hard-coded notifications to the default namespace. If the Operator were installed in a non-default namespace, the notifications would not work as expected. Now, Notifications work in non-default namespaces. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2060726[*BZ#2060726*])
+
+* Previously, the Compliance Operator was unable to evaluate default configurations used by kubelets, resulting in inaccurate results and false positives. Now, a new feature evaluates kubelet configuration and will report accurately. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2075041[*BZ#2075041*])
```
```suggestion
* Previously, the Compliance Operator was unable to evaluate default configurations used by kubelets, resulting in inaccurate results and false positives. This new feature evaluates the kubelet configuration and now reports accurately. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2075041[*BZ#2075041*])
```

Wiharris commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -13,6 +13,52 @@ These release notes track the development of the Compliance Operator in the {pro
+
+* Previously, the Compliance Operator would fail to fetch API resources when parsing machine configurations without Ignition specifications, results in `api-check-pods` to crash loop. Now, the Compliance Operator handles Machine Config Pools without Ignition specifications correctly. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2117268[*BZ#2117268*])
+
+* Previously, rules evaluating `modprobe` configuration would fail even after applying remediations, due to a mismatch in values for `modprobe` configuration. Now, the same values are used for `modprobe` configuration in checks and remediations, ensuring consistent results. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2117747[*BZ#2117747*])
```
```suggestion
* Previously, rules evaluating the `modprobe` configuration would fail even after applying remediations due to a mismatch in values for the `modprobe` configuration. Now, the same values are used for the `modprobe` configuration in checks and remediations, ensuring consistent results. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2117747[*BZ#2117747*])
```

jhrozek commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -13,6 +13,52 @@ These release notes track the development of the Compliance Operator in the {pro
+[id="compliance-operator-0-1-55-bug-fixes"]
+=== Bug fixes
+
+* Previously, the Compliance Operator hard-coded notifications to the default namespace. If the Operator were installed in a non-default namespace, the notifications would not work as expected. Now, Notifications work in non-default namespaces. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2060726[*BZ#2060726*])
```
The default namespace here refers to the openshift-compliance namespace. Because a namespace called "default" also exists on all OCP clusters, we might want to spell out openshift-compliance explicitly.

rhmdnd replied:
Is Notifications intended as a proper noun here?

jhrozek commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -13,6 +13,52 @@ These release notes track the development of the Compliance Operator in the {pro
+
+* Previously, the Compliance Operator was unable to evaluate default configurations used by kubelets, resulting in inaccurate results and false positives. This new feature evaluates the kubelet configuration and now reports accurately. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2075041[*BZ#2075041*])
+
+* Previously, the Compliance Operator had a set rule for speed at which evaluations were recorded, causing the `ocp4-kubelet-configure-event-creation` rule to fail even after remediations were applied. Now, the check and remediation uses the same variable to evaluate and remediate the system. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2082416[*BZ#2082416*])
```
I think spelling out the `eventRecordQPS` field of the kubeletConfig might be even better here.

jhrozek commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -13,6 +13,52 @@ These release notes track the development of the Compliance Operator in the {pro
+[id="compliance-operator-0-1-55-deprecations"]
+=== Deprecations
+
+* Specifying "Install into all namespaces in the cluster" or setting the `WATCH_NAMESPACES` environment variable to `""` no longer affects all namespaces. This change improves the Compliance Operator's memory usage.
```
I think we should expand on this. The effect on the end user is that any API resources installed in other namespaces than the one specified for installation will not work anymore. Typically this would mean that all resources would have to be created in the openshfit-compliance namespace.

jhrozek commented on modules/compliance-increasing-operator-limits.adoc:
```diff
@@ -0,0 +1,79 @@
+[id="compliance-increasing-operator-limits"]
+= Increasing Compliance Operator resource limits
+
+If the Operator has been installed through OLM, the Compliance Operator might require more memory than the default limits allow. The best way to mitigate this issue is to set custom resource limits.
```
So this is tricky. You're right that in many cases the memory issues were caused by OLM's way of installing operators that watch all namespaces by default. But the memory issues are not all caused strictly by this, so I wonder if we should rephrase the first sentence to say something more generic like "In some cases, the Compliance Operator might require more memory than the default limits allow".

jhrozek commented on modules/compliance-increasing-operator-limits.adoc:
```diff
@@ -0,0 +1,79 @@
++
+[IMPORTANT]
+====
+The `nodeselector` field only sets the limit for the compliance-operator
```
nodeselector?

jhrozek replied:
in addition could we cross link from here to [modules/compliance-scansetting-cr.adoc](https://github.com/openshift/openshift-docs/pull/50778/commits/10394566780a77840128b676759ae92e7c3ce349#diff-075c93655f955ab40ddb30fba38650fd721c60b7e0f3ac04447dafba58600a1b)
we already have a link from there to here, which is awesome

jhrozek commented on modules/compliance-increasing-operator-limits.adoc:
```diff
@@ -0,0 +1,79 @@
+
+== NodeSelector
+
+* The `nodeSelector` field defines a NodeSelector for the Pod created by OLM. Inject nodeSelector key-values pairs:
```
is there a reason to document the nodeselector here?

jhrozek commented on modules/compliance-applying-resource-requests-and-limits.adoc:
```diff
@@ -0,0 +1,22 @@
+
+:_content-type: CONCEPT
+[id="compliance-applying-resource-requests-and-limits"]
+= Applying resource requests and limits
```
This is all correct and useful information, but I wonder if instead of being part of the CO docs we could have all this info in some generic OCP doc? All that is described here applies equally well outside the CO world..

sheriff-rh replied:
That is definitely part of the plan. William and I are part of the larger Operator and OSDK docs team as well. The plan, for now, is to get this content available to CO users while we determine where it can also be visible in the Operator/OSDK docs too. Ultimately, it will be a one-line change to a file to insert this module into the Operator/OSDK docs when the time comes!

jhrozek commented on modules/compliance-custom-node-pools.adoc:
```diff
@@ -0,0 +1,93 @@
+
+:_content-type: PROCEDURE
+[id="compliance-custom-node-pools"]
+= Creating custom node pools
```
This section might be called Scanning custom node pools or similar (we're not creating them here..)

jhrozek commented on modules/compliance-custom-node-pools.adoc:
```diff
@@ -0,0 +1,93 @@
+
+.Procedure
+
+. To check configuration against all pools in an example cluster containing `master`, `worker`, and `infra` node pools, set the value of `ocp-var-role-master` and `opc-var-role-worker` to `infra` in the `TailoredProfile`:
```
This all looks good to me but @Vincent056 might want to double check.

jhrozek reviewed (commented):
Looking really good! I left a bunch of small comments inline.

rhmdnd commented on modules/compliance-evaluate-kubeletconfig-rules.adoc:
```diff
@@ -0,0 +1,13 @@
+
+{product-title} infrastructure may contain incomplete configuration files at run time, and nodes will assume default configuration values for missing configuration options. Some configuration options can be passed as command line arguments. As a result, the Compliance Operator cannot verify if the configuration file on the node is complete, as it may be missing options used in the rule checks.
+
+To prevent false negative results where the default configuration value passes a check, the Compliance Operator uses the `Node/Proxy` API to fetch the configuration for each node, then evaluates the properties for each node and stores a "consistent" copy to be evaluated against the rules. This increases the accuracy of the scan results. The results are then stored in `/kubeletconfig/role/<role>`.
```
I wonder if "consistent" is going to confuse people, since we're not explicit about what we're comparing to for consistency.

Perhaps:

```
... to fetch the configuration for each node in a node pool,
then all configuration options that are consistent across
nodes in the node pool are stored in a file that represents
the configuration for all nodes within that node pool.
```

I'm sure this could be worded better. But I was mainly trying to highlight what's being compared (without getting too far off into the weeds).

rhmdnd commented on modules/compliance-priorityclass.adoc:
```diff
@@ -0,0 +1,53 @@
+[id="compliance-priorityclass_{context}"]
+= Setting `PriorityClass` for `ScanSetting` scans
+
+In large scale environments, the default `PriorityClass` can be too low to guarantee Pods execute scans on time. The Compliance Operator requires a high priority to maintain compliance in a cluster.
```
nit: I'm probably splitting hairs here. Technically, the compliance operator doesn't need `PriorityClass` to function properly. Although, it would make sense for users who value compliance to set the `PriorityClass` for the operator, it's not expected.

I wouldn't want someone to interpret this as a technical dependency, as opposed to something that just makes sense for environments who place a high priority on compliance.

Since we're already leading the sentence with a situation, we could expand on it:

```
In large scale environments, the default `PriorityClass` can
be too low to guarantee Pods execute scans on time. For
clusters that must maintain compliance or guarantee
automated scanning, we recommend setting the `PriorityClass`
to ensure the Compliance Operator is always given priority
in resource constrained situations.
```

Again, I'm sure that could be worded better, but trying to make it clear the `PriorityClass` bit isn't a functional requirement of the Compliance Operator implementation, but more so a tool for deployments who consider security and compliance a requirement (e.g., I'll lose $X if my scans aren't automated, or if my compliance falls below a certain threshold.)

rhmdnd commented on modules/compliance-priorityclass.adoc:
```diff
@@ -0,0 +1,53 @@
+scanTolerations:
+  - operator: Exists
+----
+<1> If the `PriorityClass` referenced in the `ScanSetting` cannot be found, the Operator will leave the `PriorityClass` empty, issue a warning, and continue scheduling scans without a `PriorityClass`.
```
+1

rhmdnd commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -13,6 +13,52 @@ These release notes track the development of the Compliance Operator in the {pro
+[id="compliance-operator-0-1-55-new-features-and-enhancements"]
+=== New features and enhancements
+
+* `KubeletConfig` now checks the default configuration of the `KubeletConfig`. The configuration files are aggregated from all nodes into a single location. See xref:../../security/compliance_operator/compliance-operator-remediation.adoc#compliance-evaluate-kubeletconfig-rules[Evaluating `KubeletConfig` rules against default configuration values].
```
nit: into a single location per node pool?

jhrozek replied:
we should probably explicitly say that the kubelet checks changed their type from `Node` to `Platform` as part of this change. Even in internal testing, there were people who were running only node checks and thought that the kubelet checks have disappeared, see https://issues.redhat.com/browse/CMP-1591

rhmdnd commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -13,6 +13,52 @@ These release notes track the development of the Compliance Operator in the {pro
+
+* Previously, the Compliance Operator hard-coded notifications to the default namespace. If the Operator were installed in a non-default namespace, the notifications would not work as expected. Now, Notifications work in non-default namespaces. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2060726[*BZ#2060726*])
+
+* Previously, the Compliance Operator was unable to evaluate default configurations used by kubelets, resulting in inaccurate results and false positives. This new feature evaluates the kubelet configuration and now reports accurately. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2075041[*BZ#2075041*])
```
Do we have the ability to link `This new feature` to the Kubelet configuration docs?

rhmdnd commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -13,6 +13,52 @@ These release notes track the development of the Compliance Operator in the {pro
+
+* Previously, the Compliance Operator used an older version of `operator-sdk` that caused alerts about deprecated functionality. Now, an updated version of `operator-sdk` is included and there are no more alerts for deprecated functionality. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2098581[*BZ#2098581*])
+
+* Previously, the Compliance Operator made assumptions about machine configuration names, and as a result was unable to determine the relationship between machine configurations and kubelet configurations. Now, the Compliance Operator has improved handling of the machine configurations and is able to determine if a kubelet configuration is a subset of a machine configuration. (link:https://bugzil ...
```
This one does a good job describing the technical details, but I wonder if we should include what made this a bug. As a reader, I'm not sure how this affects my usage of the Compliance Operator (e.g., I should have been more clear in my doc text!)

Perhaps something like:

```
Previously, the Compliance Operator would fail to apply remediations if it couldn't
determine the relationship between kubelet and machine configurations....
```

rhmdnd reviewed (changes requested):
Great work distilling a bunch of fixes and functionality into concise prose!

Some suggestions and questions inline.

sheriff-rh commented:
@Vincent056  please review when you get a chance. Thanks!

Vincent056 commented on modules/compliance-custom-node-pools.adoc:
```diff
@@ -0,0 +1,93 @@
+  name: cis-infra-tp
+spec:
+  extends: ocp4-cis
+  title: My modified nist profile with a custom value
```
Maybe using `My modified NIST profile to scan infra node` is better here

sheriff-rh replied:
As above, `My modified NIST profile to scan example node` sounds good to me.

Vincent056 commented on modules/compliance-custom-node-pools.adoc:
```diff
@@ -0,0 +1,93 @@
+profiles:
+- apiGroup: compliance.openshift.io/v1alpha1
+  kind: Profile
+  name: ocp4-cis
```
`cis-infra-tp` will only scan Infra node, and `cop-cis` will scan `master` and `worker` node

rhmdnd replied:
What does `ocp4-cis-node` scan?

Vincent056 replied:
ocp4-cis-node is for cis-node profile, we have one kubelet rules that is in cis-node profile, ex. `kubelet_enable_protect_kernel_defaults`

Vincent056 commented on modules/compliance-custom-node-pools.adoc:
```diff
@@ -0,0 +1,93 @@
+
+.Verification
+
+* You can verify if a rule is checked through the `Node/Proxy` object by checking if `valuesUsed` of the `ComplianceCheckResult` contains `ocp4-var-role-master` or `ocp4-var-role-worker`:
```
Had a second look on this, I think this is invalid, can we change this to 

All the Platform KubeletConfig rules are checked through the `Node/Proxy` object, you can find those rules by:
```
oc get rules -o json | jq '.items[] | select(.checkType == "Platform") | select(.metadata.name | contains("ocp4-kubelet-")) | .metadata.name'
```

Vincent056 commented on modules/compliance-custom-node-pools.adoc:
```diff
@@ -0,0 +1,93 @@
++
+[source,terminal]
+----
+$ oc get ccr -n openshift-compliance -o yaml | jq '.items[] | select(.valuesUsed | contains("ocp4-var-role-master") or contains("ocp4-var-role-worker"))'`
```
same here as above

Vincent056 commented on modules/compliance-custom-node-pools.adoc:
```diff
@@ -0,0 +1,93 @@
+
+.Procedure
+
+. To check configuration against all pools in an example cluster containing `master`, `worker`, and `infra` node pools, set the value of `ocp-var-role-master` and `opc-var-role-worker` to `infra` in the `TailoredProfile`:
```
Yes, it is just an example of custom node pool

Vincent056 replied:
Maybe we should say, to scan not only master and worker node, but we scan infra node by using a tailored profile.

sheriff-rh replied:
Ah okay - I can change infra to generic "example" node to avoid confusion.

Vincent056 reviewed (changes requested):
Just some inline comments, others look good! thanks for writing the release note

sheriff-rh commented:
Pull request updated with latest feedback from engineering. Thank you @Wiharris @Vincent056 @rhmdnd and @jhrozek!

sheriff-rh commented:
/assign sheriff-rh

sheriff-rh commented:
/label peer-review-needed

kelbrown20 commented:
/label peer-review-in-progress

kelbrown20 commented on modules/compliance-increasing-operator-limits.adoc:
```diff
@@ -0,0 +1,58 @@
+
+In some cases, the Compliance Operator might require more memory than the default limits allow. The best way to mitigate this issue is to set custom resource limits.
+
+To increase the default memory and CPU limits of scanner pods, see xref:../../security/compliance_operator/compliance-scans.adoc#compliance-scansetting-cr[`ScanSetting` Custom resource].
```
Same thing with the `xref`

kelbrown20 commented on modules/compliance-kubeletconfig-sub-pool-remediation.adoc:
```diff
@@ -0,0 +1,14 @@
+[id="compliance-kubeletconfig-sub-pool-remediation"]
+= Remediating `KubeletConfig` sub pools
+
+* If you want to apply `KubeletConfig` remediation on xref:../../security/compliance_operator/compliance-operator-remediation.html#compliance-operator-apply-remediation-for-customized-mcp[`MachineConfigPool`] sub-pools, add a label to the sub-pool `MachineConfigPool`:
```
```suggestion
* If you want to apply `KubeletConfig` remediation on xref:../../security/compliance_operator/compliance-operator-remediation.html#compliance-operator-apply-remediation-for-customized-mcp[`MachineConfigPool`] sub-pools, add a label to the sub-pool `MachineConfigPool` CR:
```
Also the `xref`

kelbrown20 commented on modules/compliance-priorityclass.adoc:
```diff
@@ -0,0 +1,53 @@
+
+.Procedure
+* Set the `PriorityClass` in the `ScanSetting`:
+
```
Did you want this code block to be alighted with the procedure text?

kelbrown20 commented on modules/compliance-priorityclass.adoc:
```diff
@@ -0,0 +1,53 @@
+[id="compliance-priorityclass_{context}"]
+= Setting `PriorityClass` for `ScanSetting` scans
+
+In large scale environments, the default `PriorityClass` can be too low to guarantee Pods execute scans on time. For clusters that must maintain compliance or guarantee automated scanning, it is recommended to set the `PriorityClass` variable to ensure the Compliance Operator is always given priority in resource constrained situations.
```
```suggestion
In large scale environments, the default `PriorityClass` object can be too low to guarantee Pods execute scans on time. For clusters that must maintain compliance or guarantee automated scanning, it is recommended to set the `PriorityClass` variable to ensure the Compliance Operator is always given priority in resource constrained situations.
```

kelbrown20 commented on modules/compliance-scansetting-cr.adoc:
```diff
@@ -0,0 +1,16 @@
+[id="compliance-scansetting-cr"]
+= `ScanSetting` Custom Resource
+
+The `ScanSetting` Custom Resource now allows you to override the default CPU and memory limits of scanner pods through the scan limits attribute. The compliance operator will use defaults of 500Mi memory, 100m CPU for the scanner container, and 200Mi memory with 100m CPU for the api-resource-collector container. To set the memory limits of the operator, modify the `Subscription` object if install ...
```
Should "api-resource-collector" have back tics? Ignore if I'm wrong here!

kelbrown20 commented on modules/compliance-scansetting-cr.adoc:
```diff
@@ -0,0 +1,16 @@
+
+The `ScanSetting` Custom Resource now allows you to override the default CPU and memory limits of scanner pods through the scan limits attribute. The compliance operator will use defaults of 500Mi memory, 100m CPU for the scanner container, and 200Mi memory with 100m CPU for the api-resource-collector container. To set the memory limits of the operator, modify the `Subscription` object if install ...
+
+To increase the default CPU and memory limits of the Compliance Operator, see xref:../../security/compliance_operator/compliance-operator-troubleshooting.html#compliance-increasing-operator-limits[Increasing Compliance Operator resource limits].
```
Same thing with the `xref`

kelbrown20 commented on modules/compliance-scheduling-pods-with-resource-requests.adoc:
```diff
@@ -0,0 +1,59 @@
+[id="compliance-scheduling-pods-with-resource-requests"]
+= Scheduling Pods with resource requests
+
+When a Pod is created, the scheduler selects a Node for the Pod to run on. Each node has a maximum capacity for each resource type in the amount of CPU and memory it can provide for the Pods. The scheduler ensures that the sum of the resource requests of the scheduled containers is less than the capacity nodes for each resource type.
```
I don't think `Node` needs to be capitalized here

kalexand-rh replied:
+1

kelbrown20 commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -13,6 +13,53 @@ These release notes track the development of the Compliance Operator in the {pro
+
+* The `ScanSetting` Custom Resource now allows users to override the default CPU and memory limits of scanner pods through the `scanLimits` attribute.  For more information, see xref:../../security/compliance_operator/compliance-operator-troubleshooting.adoc#compliance-increasing-operator-limits[Increasing Compliance Operator resource limits].
+
+* A `PriorityClass` can now be set through `ScanSetting`. This will ensure the Compliance Operator is prioritized and minimizes the chance that the cluster will fall out of compliance. For more information, see xref:../../security/compliance_operator/compliance-operator-advanced.adoc#compliance-priorityclass_compliance-advanced[Setting `PriorityClass` for `ScanSetting` scans].
```
```suggestion
* A `PriorityClass` object can now be set through a`ScanSetting` CR. This will ensure the Compliance Operator is prioritized and minimizes the chance that the cluster will fall out of compliance. For more information, see xref:../../security/compliance_operator/compliance-operator-advanced.adoc#compliance-priorityclass_compliance-advanced[Setting `PriorityClass` for `ScanSetting` scans].
```

kelbrown20 commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -13,6 +13,53 @@ These release notes track the development of the Compliance Operator in the {pro
+
+* Previously, the Compliance Operator would not clean up pods used to scan infrastructure when using the `debug=true` option in the scan settings. This caused pods to be left on the cluster even after deleting the Scan Setting Binding. Now, pods are always deleted when a Scan Setting Binding is deleted.(link:https://bugzilla.redhat.com/show_bug.cgi?id=2092913[*BZ#2092913*])
+
+* Previously, the Compliance Operator used an older version of `operator-sdk` that caused alerts about deprecated functionality. Now, an updated version of `operator-sdk` is included and there are no more alerts for deprecated functionality. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2098581[*BZ#2098581*])
```
```suggestion
* Previously, the Compliance Operator used an older version of the `operator-sdk` command that caused alerts about deprecated functionality. Now, an updated version of the `operator-sdk` command is included and there are no more alerts for deprecated functionality. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2098581[*BZ#2098581*])
```

kelbrown20 reviewed (commented):
Hey Andrew! Great work! A couple suggestions and questions, there were also some xrefs that we in modules. Please let me know if you have any questions or concerns about anything that I added!

kelbrown20 commented:
/remove-label peer-review-needed
/remove-label peer-review-in-progress
/label peer-review-done

sheriff-rh commented:
Addressed peer review feedback and updated refs, then pushed latest. This content is now ready for merge upon release.

sheriff-rh commented:
Updated references and changed version number to v0.1.56; pushing updated build preview to existing links.

sheriff-rh commented on modules/compliance-supported-profiles.adoc:
```diff
@@ -86,12 +86,14 @@ The Compliance Operator provides the following compliance profiles:
 |0.1.47+
 |link:https://www.pcisecuritystandards.org/document_library?document=pci_dss[PCI Security Standards &#174; Council Document Library]
 |`x86_64`
+ `ppc64le`
```
@prb112 As a part of the next downstream Compliance Operator release (0.1.57) I have included Power support to the PCI profiles as previously discussed. You can view a preview [here](http://file.rdu.redhat.com/antaylor/co-0.1.55-rns/security/compliance_operator/compliance-operator-supported-profiles.html#compliance-supported-profiles_compliance-operator-supported-profiles). Please let me know if I need to make any further changes!

sheriff-rh replied:
@rhmdnd for visibility ^

prb112 replied:
awesome - thank you

xiaojiey commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -13,6 +13,53 @@ These release notes track the development of the Compliance Operator in the {pro
+[id="compliance-operator-0-1-57-deprecations"]
+=== Deprecations
+
+* Specifying "Install into all namespaces in the cluster" or setting the `WATCH_NAMESPACES` environment variable to `""` no longer affects all namespaces. Any API resources installed in namespaces not specified at the time of Compliance Operator installation will no longer be operational. API resources might require creation in the `openshift-compliance` namespace. This change improves the Compli ...
```
API resources might require creation in the `openshift-compliance` namespace.
It could be updated to 
`API resources might require creation in the selected namespace, `openshift-compliance` by default.`
As user might install it to a namespace other than `openshift-compliance`.

xiaojiey commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -13,6 +13,53 @@ These release notes track the development of the Compliance Operator in the {pro
+* A `PriorityClass` object can now be set through `ScanSetting`. This will ensure the Compliance Operator is prioritized and minimizes the chance that the cluster will fall out of compliance. For more information, see xref:../../security/compliance_operator/compliance-operator-advanced.adoc#compliance-priorityclass_compliance-advanced[Setting `PriorityClass` for `ScanSetting` scans].
+
+[id="compliance-operator-0-1-57-bug-fixes"]
+=== Bug fixes
```
For the bug fixes, it only highlight the bugs from bugzilla. Do we need to add the bugs from Jira issues? Thanks.

sheriff-rh replied:
@rhmdnd can you confirm if any of the Jira issues need to be added to the bug fix section of this release? It seems like [OCPBUGS-2331](https://issues.redhat.com/browse/OCPBUGS-2331) and [OCPBUGS-1066](https://issues.redhat.com/browse/OCPBUGS-1066) are bug fixes since the last downstream release, correct? Can you assist with the cause/consequence/fix/result in the Jiras so I can update the bug fixes for this release?

sheriff-rh replied:
We may have to address additional bugs post-release, as this content is now live and needs to be merged. The errata points to this documentation.

sheriff-rh commented:
/label merge-review-needed

kalexand-rh commented on modules/compliance-custom-node-pools.adoc:
```diff
@@ -0,0 +1,94 @@
+schedule: '0 1 * * *'
+----
+
+. Create a scan using `ScanSettingBinding` CR:
```
> ! using noun
To avoid ambiguity, replace this gerund with either "by using" or "that uses".

kalexand-rh commented on modules/compliance-custom-node-pools.adoc:
```diff
@@ -0,0 +1,94 @@
+  name: default
+----
+
+The Compliance Operator checks the runtime `KubeletConfig` through the `Node/Proxy` API object, and then uses variables such as `ocp-var-role-master` and `ocp-var-role-worker` to determine the nodes it performs the check against. In the `ComplianceCheckResult`, the `KubeletConfig` rules will be shown as `ocp4-cis-kubelet-*`. The scan only passes if all selected nodes pass this check.
```
```suggestion
The Compliance Operator checks the runtime `KubeletConfig` through the `Node/Proxy` API object and then uses variables such as `ocp-var-role-master` and `ocp-var-role-worker` to determine the nodes it performs the check against. In the `ComplianceCheckResult`, the `KubeletConfig` rules are shown as `ocp4-cis-kubelet-*`. The scan passes only if all selected nodes pass this check.
```

kalexand-rh commented on modules/compliance-custom-node-pools.adoc:
```diff
@@ -0,0 +1,94 @@
++
+[source,terminal]
+----
+$ oc get rules -o json | jq '.items[] | select(.checkType == "Platform") | select(.metadata.name | contains("ocp4-kubelet-")) | .metadata.name'
```
> Do not use jq in commands (unless it is truly required), because this requires users to install the jq tool.  Oftentimes, the same or similar result can be accomplished using `jsonpath` for `oc` commands.

This is out of scope for a merge review, but consider fixing it in the future.

kalexand-rh commented on modules/compliance-evaluate-kubeletconfig-rules.adoc:
```diff
@@ -0,0 +1,13 @@
+[id="compliance-evaluate-kubeletconfig-rules"]
+= Evaluating `KubeletConfig` rules against default configuration values
+
+{product-title} infrastructure may contain incomplete configuration files at run time, and nodes will assume default configuration values for missing configuration options. Some configuration options can be passed as command line arguments. As a result, the Compliance Operator cannot verify if the configuration file on the node is complete, as it may be missing options used in the rule checks.
```
> ! may verb
Use only to indicate permission. Otherwise, use "can" to indicate ability, or use "might" to indicate possibility.
```suggestion
{product-title} infrastructure might contain incomplete configuration files at run time, and nodes assume default configuration values for missing configuration options. Some configuration options can be passed as command line arguments. As a result, the Compliance Operator cannot verify if the configuration file on the node is complete because it might be missing options used in the rule checks.
```

kalexand-rh commented on modules/compliance-evaluate-kubeletconfig-rules.adoc:
```diff
@@ -0,0 +1,13 @@
+
+{product-title} infrastructure may contain incomplete configuration files at run time, and nodes will assume default configuration values for missing configuration options. Some configuration options can be passed as command line arguments. As a result, the Compliance Operator cannot verify if the configuration file on the node is complete, as it may be missing options used in the rule checks.
+
+To prevent false negative results where the default configuration value passes a check, the Compliance Operator uses the `Node/Proxy` API to fetch the configuration for each node in a node pool, then all configuration options that are consistent across nodes in the node pool are stored in a file that represents the configuration for all nodes within that node pool. This increases the accuracy of  ...
```
> Generally, when you specify the name of a command, parameter, or option in text, specify the word command, parameter, or option after the name.

Command, parameter, and option names are not translated; without the identifying noun, users might be confused about what the name represents.

kalexand-rh commented on modules/compliance-increasing-operator-limits.adoc:
```diff
@@ -0,0 +1,60 @@
+
+.Procedure
+
+. To increase the Operator's memory limits to 500Mi, create the following patch file:
```
\`500Mi` or 500 Mi

Check out the "Units of measurement" section of the ISG.

I see a few other places in this PR with this issue.

kalexand-rh commented on modules/compliance-increasing-operator-limits.adoc:
```diff
@@ -0,0 +1,60 @@
++
+[source,terminal]
+----
+$ oc patch sub compliance-operator -nopenshift-compliance --patch-file co-memlimit-patch.yaml --type=merge
```
You don't specify the file name in the previous step, so it would be clearer to use a variable for the file name here. Consider making that change at some point.

kalexand-rh commented on modules/compliance-increasing-operator-limits.adoc:
```diff
@@ -0,0 +1,60 @@
+
+== Resources
+
+The `resources` field defines Resource Constraints for all the containers in the Pod created by OLM.
```
You should expand the first use of OLM. And it looks like "pod" shouldn't be capitalized [here](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#general-references).

kalexand-rh commented on modules/compliance-increasing-operator-limits.adoc:
```diff
@@ -0,0 +1,60 @@
+$ oc patch sub compliance-operator -nopenshift-compliance --patch-file co-memlimit-patch.yaml --type=merge
+----
+
+== Resources
```
Ideally, this content would be in a separate reference module.

kalexand-rh commented on modules/compliance-kubeletconfig-sub-pool-remediation.adoc:
```diff
@@ -0,0 +1,14 @@
+[id="compliance-kubeletconfig-sub-pool-remediation"]
+= Remediating `KubeletConfig` sub pools
+
+* If you want to apply `KubeletConfig` remediation on `MachineConfigPool` sub-pools, add a label to the sub-pool `MachineConfigPool` CR:
```
Consider making this module follow the [procedure](https://raw.githubusercontent.com/redhat-documentation/modular-docs/master/modular-docs-manual/files/TEMPLATE_PROCEDURE_doing-one-procedure.adoc) template.

kalexand-rh commented on modules/compliance-priorityclass.adoc:
```diff
@@ -0,0 +1,54 @@
+
+.Procedure
+
+* Set the `PriorityClass` in the `ScanSetting`:
```
> Generally, when you specify the name of a command, parameter, or option in text, specify the word command, parameter, or option after the name.

kalexand-rh commented on modules/compliance-scansetting-cr.adoc:
```diff
@@ -0,0 +1,16 @@
+[id="compliance-scansetting-cr"]
+= `ScanSetting` Custom Resource
+
+The `ScanSetting` Custom Resource now allows you to override the default CPU and memory limits of scanner pods through the scan limits attribute. The compliance operator will use defaults of 500Mi memory, 100m CPU for the scanner container, and 200Mi memory with 100m CPU for the `api-resource-collector` container. To set the memory limits of the operator, modify the `Subscription` object if insta ...
```
Check you capitalization here.

kalexand-rh commented on modules/compliance-scansetting-cr.adoc:
```diff
@@ -0,0 +1,16 @@
+
+[IMPORTANT]
+====
+Increasing the memory limit for the Compliance Operator or the scanner pods is needed if the default limits are not sufficient and the Operator or scanner pods are killed by the Out Of Memory (OOM) killer process.
```
> 🚫  kill verb
Do not use except when you are documenting a UNIX system, such as AIX. Use "end" or "stop".

kalexand-rh commented on modules/compliance-scheduling-pods-with-resource-requests.adoc:
```diff
@@ -0,0 +1,59 @@
+
+Although memory or CPU resource usage on nodes is very low, the scheduler might still refuse to place a Pod on a node if the capacity check fails to protect against a resource shortage on a node.
+
+For each container, you can specify resource limits and requests, including the following:
```
Is this the entire list? If so, "For each container, you can specify the following resource limits and request:"

Either way, you need a noun after "following."

kalexand-rh commented on modules/compliance-scheduling-pods-with-resource-requests.adoc:
```diff
@@ -0,0 +1,59 @@
+
+[NOTE]
+====
+Although you can only specify requests and limits for individual containers, it is also useful to consider the overall resource requests and limits for a Pod. For a particular resource, a Pod resource request or limit is the sum of the resource requests or limits of that type for each container in the Pod.
```
```suggestion
Although you can specify requests and limits for only individual containers, it is also useful to consider the overall resource requests and limits for a pod. For a particular resource, a pod resource request or limit is the sum of the resource requests or limits of that type for each container in the pod.
```

kalexand-rh commented on modules/compliance-scheduling-pods-with-resource-requests.adoc:
```diff
@@ -0,0 +1,59 @@
+        memory: "128Mi"
+        cpu: "500m"
+----
+<1> The Pod is requesting 64Mi of memory and 250 millicores.
```
Please be consistent with abbreviating and not abbreviating units in the same sentence. Also, you need a noun after "250 millicores"

kalexand-rh commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -13,6 +13,53 @@ These release notes track the development of the Compliance Operator in the {pro
+
+* Previously, the Compliance Operator hard-coded notifications to the default `openshift-compliance` namespace. If the Operator were installed in a non-default namespace, the notifications would not work as expected. Now, notifications work in non-default `openshift-compliance` namespaces. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2060726[*BZ#2060726*])
+
+* Previously, the Compliance Operator was unable to evaluate default configurations used by kubelet objectss, resulting in inaccurate results and false positives. xref:../../security/compliance_operator/compliance-operator-remediation.adoc#compliance-evaluate-kubeletconfig-rules[This new feature] evaluates the kubelet configuration and now reports accurately. (link:https://bugzilla.redhat.com/sho ...
```
```suggestion
* Previously, the Compliance Operator was unable to evaluate default configurations used by kubelet objects, resulting in inaccurate results and false positives. xref:../../security/compliance_operator/compliance-operator-remediation.adoc#compliance-evaluate-kubeletconfig-rules[This new feature] evaluates the kubelet configuration and now reports accurately. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2075041[*BZ#2075041*])
```

kalexand-rh commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -13,6 +13,53 @@ These release notes track the development of the Compliance Operator in the {pro
+
+* Previously, the Compliance Operator was unable to evaluate default configurations used by kubelet objectss, resulting in inaccurate results and false positives. xref:../../security/compliance_operator/compliance-operator-remediation.adoc#compliance-evaluate-kubeletconfig-rules[This new feature] evaluates the kubelet configuration and now reports accurately. (link:https://bugzilla.redhat.com/sho ...
+
+* Previously, the Compliance Operator reported the `ocp4-kubelet-configure-event-creation` rule in a `FAIL` state after applying an automatic remediation due to the `eventRecordQPS` value was set higher than the default value. Now, the `ocp4-kubelet-configure-event-creation` rule remediation sets the default value and the rule applies correctly. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2 ...
```
```suggestion
* Previously, the Compliance Operator reported the `ocp4-kubelet-configure-event-creation` rule in a `FAIL` state after applying an automatic remediation because the `eventRecordQPS` value was set higher than the default value. Now, the `ocp4-kubelet-configure-event-creation` rule remediation sets the default value, and the rule applies correctly. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2082416[*BZ#2082416*])
```

kalexand-rh commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -13,6 +13,53 @@ These release notes track the development of the Compliance Operator in the {pro
+
+* Previously, the Compliance Operator would not clean up pods used to scan infrastructure when using the `debug=true` option in the scan settings. This caused pods to be left on the cluster even after deleting the Scan Setting Binding. Now, pods are always deleted when a Scan Setting Binding is deleted.(link:https://bugzilla.redhat.com/show_bug.cgi?id=2092913[*BZ#2092913*])
+
+* Previously, the Compliance Operator used an older version of the `operator-sdk` command that caused alerts about deprecated functionality. Now, an updated version of the `operator-sdk` command is included and there are no more alerts for deprecated functionality. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2098581[*BZ#2098581*])
```
```suggestion
* Previously, the Compliance Operator used an older version of the `operator-sdk` command that caused alerts about deprecated functionality. Now, an updated version of the `operator-sdk` command is included, and there are no more alerts for deprecated functionality. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2098581[*BZ#2098581*])
```

kalexand-rh commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -13,6 +13,53 @@ These release notes track the development of the Compliance Operator in the {pro
+
+* The `ocp4-configure-network-policies` rule requires manual intervention to perform effectively. New descriptive instructions and rule updates increase applicability of the `ocp4-configure-network-policies` rule for clusters using Calico CNIs. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2091794[*BZ#2091794*])
+
+* Previously, the Compliance Operator would not clean up pods used to scan infrastructure when using the `debug=true` option in the scan settings. This caused pods to be left on the cluster even after deleting the Scan Setting Binding. Now, pods are always deleted when a Scan Setting Binding is deleted.(link:https://bugzilla.redhat.com/show_bug.cgi?id=2092913[*BZ#2092913*])
```
Should "Scan Setting Binding" be capitalized here?

kalexand-rh commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -13,6 +13,53 @@ These release notes track the development of the Compliance Operator in the {pro
+* Previously, the Compliance Operator would fail to apply remediations if it could not
+determine the relationship between kubelet and machine configurations. Now, the Compliance Operator has improved handling of the machine configurations and is able to determine if a kubelet configuration is a subset of a machine configuration. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2102511[*BZ#2102511*])
+
+* Previously, The rule for `ocp4-cis-node-master-kubelet-enable-cert-rotation` did not properly describe success criteria. As a result, the requirements for `RotateKubeletClientCertificate` were unclear. Now, the rule for `ocp4-cis-node-master-kubelet-enable-cert-rotation` reports accurately regardless of the configuration present in the kubelet configuration file. (link:https://bugzilla.redhat.c ...
```
```suggestion
* Previously, the rule for `ocp4-cis-node-master-kubelet-enable-cert-rotation` did not properly describe success criteria. As a result, the requirements for `RotateKubeletClientCertificate` were unclear. Now, the rule for `ocp4-cis-node-master-kubelet-enable-cert-rotation` reports accurately regardless of the configuration present in the kubelet configuration file. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2105153[*BZ#2105153*])
```

kalexand-rh commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -13,6 +13,53 @@ These release notes track the development of the Compliance Operator in the {pro
+* Previously, the Compliance Operator used an older version of the `operator-sdk` command that caused alerts about deprecated functionality. Now, an updated version of the `operator-sdk` command is included and there are no more alerts for deprecated functionality. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2098581[*BZ#2098581*])
+
+* Previously, the Compliance Operator would fail to apply remediations if it could not
+determine the relationship between kubelet and machine configurations. Now, the Compliance Operator has improved handling of the machine configurations and is able to determine if a kubelet configuration is a subset of a machine configuration. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2102511[*BZ#2102511*])
```
You don't need this hard wrap.

kalexand-rh commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -13,6 +13,53 @@ These release notes track the development of the Compliance Operator in the {pro
+
+* Previously, the rule for checking idle streaming timeouts did not consider default values, resulting in inaccurate rule reporting. Now, more robust checks ensure increased accuracy in results based on default configuration values. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2105878[*BZ#2105878*])
+
+* Previously, the Compliance Operator would fail to fetch API resources when parsing machine configurations without Ignition specifications, results in `api-check-pods` to crash loop. Now, the Compliance Operator handles Machine Config Pools without Ignition specifications correctly. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2117268[*BZ#2117268*])
```
```suggestion
* Previously, the Compliance Operator would fail to fetch API resources when parsing machine configurations without Ignition specifications, which caused the `api-check-pods` <you need a noun here> to crash loop. Now, the Compliance Operator handles Machine Config Pools that do not have Ignition specifications correctly. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2117268[*BZ#2117268*])
```

kalexand-rh commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -13,6 +13,53 @@ These release notes track the development of the Compliance Operator in the {pro
+[id="compliance-operator-0-1-57-deprecations"]
+=== Deprecations
+
+* Specifying "Install into all namespaces in the cluster" or setting the `WATCH_NAMESPACES` environment variable to `""` no longer affects all namespaces. Any API resources installed in namespaces not specified at the time of Compliance Operator installation will no longer be operational. API resources might require creation in the selected namespace, or the `openshift-compliance` namespace by de ...
```
What is "Install into all namespaces in the cluster"? If it's a UI option, it should be enclosed in [asterisks](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#quick-markup-reference).

sheriff-rh commented:
Alright @kalexand-rh I addressed all feedback and fixed related build issues. I believe we are ready for merge now. 

Thank you for all of those comments, I'm still amazed at how much our standards and processes have changed, but also by how much slipped through previously. Thank you again!
/label merge-review-needed

sheriff-rh commented:
@mburke5678  this is urgent for a newly released errata that refers to the content in this pull request. Please merge as soon as possible. Thank you!

sheriff-rh commented:
/cherrypick enterprise-4.12

sheriff-rh commented:
/cherrypick enterprise-4.11

sheriff-rh commented:
/cherrypick enterprise-4.10

sheriff-rh commented:
/cherrypick enterprise-4.9

sheriff-rh commented:
@kalexand-rh cherry picks are ready for merge when you have a moment. Thank you!
